From 50f5f3d1f6feb50d9a9b2f24b858f5cad16aef42 Mon Sep 17 00:00:00 2001
From: Dominique Martinet <asmadeus@codewreck.org>
Date: Sat, 26 Aug 2023 09:31:46 +0900
Subject: [PATCH 2/2] don't ignore config's httpAddress

- safe/unsafe port would listen to node's default (0.0.0.0) which is probably
not intended. Note this change can break existing setups as the default value
changes to 127.0.0.1 with this.
- websocket listener hardcoded localhost
---
 lib/http-worker.js | 4 ++--
 server.js          | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/http-worker.js b/lib/http-worker.js
index c3ed9754c3fc..147a7d841423 100644
--- a/lib/http-worker.js
+++ b/lib/http-worker.js
@@ -599,9 +599,9 @@ app.use(function (err, req, res /*, next*/) {
 var server = Http.createServer(app);
 
 nThen(function (w) {
-    server.listen(Env.httpPort, w());
+    server.listen(Env.httpPort, Env.httpAddress, w());
     if (Env.httpSafePort) {
-        server.listen(Env.httpSafePort, w());
+        server.listen(Env.httpSafePort, Env.httpAddress, w());
     }
     server.on('upgrade', function (req, socket, head) {
         // TODO warn admins that websockets should only be proxied in this way in a dev environment
diff --git a/server.js b/server.js
index 8c295fef9fd2..b30e77e276ac 100644
--- a/server.js
+++ b/server.js
@@ -81,7 +81,7 @@ nThen(function (w) {
         process.exit(1);
     }
     Env.httpServer = Http.createServer(app);
-    Env.httpServer.listen(Env.websocketPort, 'localhost', w(function () {
+    Env.httpServer.listen(Env.websocketPort, Env.httpAddress, w(function () {
         Env.Log.info('WEBSOCKET_LISTENING', {
             port: Env.websocketPort,
         });
-- 
2.41.0

